### ISSUE TITLE
blueman-manager crashes when bluetoothd is restarted 

### ISSUE THREAD

**Issue by Yannik** [blueman-project/blueman#1231]
**Environment**

blueman: `1:2.1.2-1.fc31`
BlueZ: `5.54-1.fc31`
Distribution: Fedora 31
Desktop environment: gnome

**The issue**

When restarting bluetoothd (`systemctl restart bluetooth.service`), `blueman-manager` crashes:
```
...
blueman-manager 12.42.06 DEBUG    Base:74 do_g_properties_changed: /org/bluez/hci0 {'Powered': False, 'Discovering': False, 'Class': 0}
blueman-manager 12.42.06 DEBUG    Manager:66 _on_object_removed: /org/bluez/hci0/dev_0C_A6_94_7D_B3_03
blueman-manager 12.42.06 DEBUG    Manager:66 _on_object_removed: /org/bluez/hci0/dev_38_18_4C_D3_19_E8
blueman-manager 12.42.06 DEBUG    Manager:66 _on_object_removed: /org/bluez/hci0/dev_00_22_37_46_50_2E
blueman-manager 12.42.06 DEBUG    Manager:62 _on_object_removed: /org/bluez/hci0
blueman-manager 12.42.06 DEBUG    DeviceList:238 set_adapter: Setting adapter to: None
blueman-manager 12.42.06 DEBUG    ManagerToolbar:62 on_adapter_changed: toolbar adapter None
blueman-manager 12.42.06 INFO     Manager:82 on_dbus_name_vanished: org.bluez
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xf7 in position 1: invalid start byte
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xd4 in position 3: invalid continuation byte
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xd8 in position 3: invalid continuation byte
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xb0 in position 0: invalid start byte
blueman-manager 12.42.06 INFO     Manager:102 on_dbus_name_appeared: org.bluez :1.2393
blueman-manager 12.42.06 ERROR    Manager:118 on_dbus_name_appeared: Default adapter not found, trying first available.
blueman-manager 12.42.06 ERROR    Manager:122 on_dbus_name_appeared: No adapter(s) found, exiting
blueman-manager 12.42.06 INFO     PulseAudioUtils:669 _on_delete: Destroying PulseAudioUtils instance
...
```

This is especially annoying as my initramfs is locking my encrypted luks `/` volume and chrooting into the initramfs to ensure data security. To do that, it need to `systemctl stop bluetooth` before suspending, and `systemctl start bluetooth` while resuming, because bluetoothd can otherwise inhibit the suspend process as it generates i/o on the locked encrypted root volume.

This means that blueman-manager crashes after every suspend.

**Suggested solution** 

Simply show "no bluetooth adapter found" in blueman-manager.
I found [this](https://github.com/blueman-project/blueman/blob/master/blueman/main/Manager.py#L100) FIXME, which seems related to this:  
> FIXME ui can handle BlueZ start/stop but we should inform user

**Comment by infirit** [comment]
What I wanted to do (FIXME is mine) is to replace the devicelist with a frame which contains a message that either bluetooth is not available or that the adapter is not powered. In case of the latter show a button or provide instruction to power the adapter.

**Comment by Yannik** [comment]
@infirit With my patch, when no adapter is available, no devices are shown and the adapte menu is greyed out. I think this is a pretty good indication that the bluetooth adapter is not available (plus there's always the stdout/stderr output) already. Adding an extra frame like you suggested seems like a nice suggestion though. 
What about opening another issue to track this?

I'd like to keep this separate because this PR is targeted at fixing the crash on stop/start.

**Comment by cschramm** [comment]
I'm a little bit confused what's the actual issue here. Looking at #1235, #1245 should fix it, correct? How's #1234 related?

**Comment by infirit** [comment]
It started with several exception when more than one device is removed from the list.  #1234 will sort this.

The other is that crash handlers (abrt) will pickup ``exit(1)`` and report it as a crash. In our case we really expect to exit so sending exit code 1 is a bit harsh. I did  #1245 for this.

Then the one I do not agree with is  #1235 as it basically removes the error dialog en shutdown keeping blueman-manager open. This is technically no problem but I want the user to be properly informed that bluetooth is not available/blocked, ideally with a button to power the adapter.

**Comment by cschramm** [comment]
OK, so this is actually about two issues, one already resolved by #1245 and one resolved in #1234. #1235 can be closed then and this ticket can be closed when #1234 is merged.
